src/Microsoft.ML.CpuMath/Microsoft.ML.CpuMath.csproj

```diff
@@ -30,6 +30,7 @@
     <Compile Remove="AvxIntrinsics.cs" />
 
     <PackageReference Include="System.Memory" Version="$(SystemMemoryVersion)" />
+    <ProjectReference Include="..\..\src\Microsoft.ML.Core\Microsoft.ML.Core.csproj" />
```

Comment:
I don't object per se, but we've previously expended significant effort towards not doing this --that is, CPU-math we've seen is independent of anything in ML.NET itself, and we've gone through some effort to ensure that this is possible. (E.g., the whole "private contracts" thing.)

The thing is, I have absolutely no memory of why that is so, or why that was necessary. I think it even predates me working on this project, so this "desire" is maybe over five years old.

So I think it's fine, but it makes me nervous...

Reply:
Yep, I don't recall anything about this not depending on Core. Maybe it had to do with the idea that we could 'export as code' and link with CpuMath, but not M.ML?

---
In reply to: [234737871](https://github.com/dotnet/machinelearning/pull/1659#discussion_r234737871) [](ancestors = 234737871)

Reply:
This is bad. It completely breaks CpuMath as a standalone package. We need to fix.

CpuMath ships in its own NuGet package that `Microsoft.ML` depends on. This reference is backwards.

See

[https://www.nuget.org/packages/Microsoft.ML.CpuMath/](https://www.nuget.org/packages/Microsoft.ML.CpuMath/)
[https://www.nuget.org/packages/Microsoft.ML/](https://www.nuget.org/packages/Microsoft.ML/)


---
In reply to: [234822215](https://github.com/dotnet/machinelearning/pull/1659#discussion_r234822215) [](ancestors = 234822215,234737871)

Reply:
I've opened https://github.com/dotnet/machinelearning/issues/1688 for this issue.